tests/system/stats/stackdriver/stackdriver_stats_test.py

```diff
@@ -0,0 +1,121 @@
+
+import os
+import random
+import time
+import unittest
+from pprint import pprint
+from opencensus.stats import aggregation as aggregation_module
+from opencensus.stats.exporters import stackdriver_exporter as stackdriver
+from opencensus.stats import measure as measure_module
+from opencensus.stats import stats as stats_module
+from opencensus.stats import view as view_module
+from opencensus.tags import tag_key as tag_key_module
+from opencensus.tags import tag_map as tag_map_module
+from opencensus.tags import tag_value as tag_value_module
+from opencensus.common.transports import sync
+from google.cloud import monitoring_v3
+
+MiB = 1 << 20
+
+PROJECT = os.environ.get('GCLOUD_PROJECT_PYTHON')
+
+
+class TestBasicStats(unittest.TestCase):
+
+    def test_stats_record_sync(self):
+        FRONTEND_KEY = tag_key_module.TagKey("SampleKeySync")
+        VIDEO_SIZE_MEASURE = measure_module.MeasureInt(
+            "SampleMeasureNameSync", "SampleDescriptionSync", "By")
+        VIDEO_SIZE_VIEW_NAME = "SampleViewNameSync"
+        VIDEO_SIZE_DISTRIBUTION = aggregation_module.DistributionAggregation(
+                                    [0.0, 16.0 * MiB, 256.0 * MiB])
+        VIDEO_SIZE_VIEW = view_module.View(VIDEO_SIZE_VIEW_NAME,
+                                        "SampleViewDescriptionSync",
+                                        [FRONTEND_KEY],
+                                        VIDEO_SIZE_MEASURE,
+                                        VIDEO_SIZE_DISTRIBUTION)
+
+        stats = stats_module.Stats()
+        view_manager = stats.view_manager
+        stats_recorder = stats.stats_recorder
+
+        client = monitoring_v3.MetricServiceClient()
+        exporter = stackdriver.StackdriverStatsExporter(options=stackdriver.Options(project_id=PROJECT),
+                                                        client=client,
+                                                        transport=sync.SyncTransport)
+        view_manager.register_exporter(exporter)
+
+        # Register view.
+        view_manager.register_view(VIDEO_SIZE_VIEW)
+
+        # Sleep for [0, 10] milliseconds to fake work.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        # Process video.
+        # Record the processed video size.
+        tag_value = tag_value_module.TagValue("1200")
+        tag_map = tag_map_module.TagMap()
+        tag_map.insert(FRONTEND_KEY, tag_value)
+        measure_map = stats_recorder.new_measurement_map()
+        measure_map.measure_int_put(VIDEO_SIZE_MEASURE, 25 * MiB)
+
+        measure_map.record(tag_map)
+
+
+        # Sleep for [0, 10] milliseconds to fake wait.
+        time.sleep(random.randint(1, 10) / 1000.0)
+
+        name = exporter.client.project_path(PROJECT)
+        list_metrics_descriptors = exporter.client.list_metric_descriptors(name)
+        element = [element for element in list_metrics_descriptors if element.description == "SampleViewDescriptionSync"]
+        self.assertTrue(any(element))
```

Comment:
Printing out the element it looks like
```
[name: "projects/[PROJECT_ID]/metricDescriptors/custom.googleapis.com/opencensus/SampleViewNameSync"
labels {
  key: "SampleKeySync"
}
metric_kind: CUMULATIVE
value_type: DISTRIBUTION
unit: "By"
description: "SampleViewDescriptionSync"
display_name: "OpenCensus/SampleViewNameSync"
type: "custom.googleapis.com/opencensus/SampleViewNameSync"
]
```
Can we assert each of the attributes equals what we expected?

And we will also need to remove the exported metrics after running the test, because next time even if the exporter fails, the test will still pass as it is using the previous data. Or we can randomly generate some unique name for the metric to make sure the one we assert is the one we just exported.